src/web-mercator-viewport.js

```diff
@@ -72,14 +77,12 @@ export default class WebMercatorViewport extends Viewport {
     farZMultiplier = 10
   } = {}) {
     // Viewport - support undefined arguments
-    width = width !== undefined ? width : DEFAULT_MAP_STATE.width;
```

Comment:
This was written without understanding that default arguments work on opts[key] === 'undefined'. I think we could drop the unusual looking `ensureFinite` that seems to be creeping in everywhere and just set the default arguments in the named parameters.

If we are going to try to ensure cases where the data is really bad, maybe throw an exception if we supply bad data rather than silently replacing it.